railties/lib/rails/generators/rails/app/templates/bin/spring.tt

```diff
@@ -9,5 +9,5 @@ if !defined?(Spring) && [nil, "development", "test"].include?(ENV["RAILS_ENV"])
     require "spring/binstub"
   rescue Gem::LoadError
     # Ignore when Spring is not installed.
-  end
+  end if Bundler.locked_gems
```

Comment:
Instead of this trailing conditional, we could chain safe navigation operators:

```ruby
Bundler.locked_gems&.specs&.find { |spec| spec.name == "spring" }&.tap do |spring|
```

I'm usually wary of long chains of safe navigation operators, but, in this case, I think they're more clear than a trailing conditional.